### PR TITLE
Add quantum-serverless run docs

### PR DESCRIPTION
**Pull request by psschwei**
Aiming to have this merged by Nov 1

### CODE REVIEW

File: docs/run/quantum-serverless.mdx
Comment (abbycross) [inline docs/run/quantum-serverless.mdx:8]:
```suggestion
Premium users can run their workloads remotely on classical compute made available through IBM Quantum Platform..
```

Comment (abbycross) [reply docs/run/quantum-serverless.mdx:8]: Spell out IQP

File: docs/run/quantum-serverless.mdx
Comment (abbycross) [inline docs/run/quantum-serverless.mdx:12]:
```suggestion
Here is an example of computing the quasi-probability distribution of a quantum circuit using the Qiskit Runtime Sampler primitive. This Python script should be saved in your working directory. (Warning! All contents of the working directory will be shipped to the cluster for execution.)
```

Comment (abbycross) [reply docs/run/quantum-serverless.mdx:12]: capitalization

File: docs/run/quantum-serverless.mdx
Comment (abbycross) [inline docs/run/quantum-serverless.mdx:6]:
```suggestion
# Run workloads with quantum serverless
```

File: docs/run/quantum-serverless.mdx
Comment (abbycross) [inline docs/run/quantum-serverless.mdx:2]:
```suggestion
title: Run workloads with quantum serverless
```

**Comment by abbycross** [comment]
I apologize for starting to add my comments while this is still in draft form! It was labeled "in review" so I just went ahead without noticing it's not necessarily "ready for prime time"... 😨

**Comment by psschwei** [comment]
No worries! I kept it in draft since I think we want to wait to merge until Nov 1 or thereabouts, but otherwise the content is more or less ready to go.

**Comment by jenglick** [comment]
todos:
- [ ] remove `QuantumServerless` and replace by the provider
- [ ] add upload step to the example
- [ ] update `Program` with final name (pending)
- [ ] update `program.py` to reflect final name (pending)

cc @icekhan13

**Comment by psschwei** [comment]
@abbycross going forward, could you leave comments rather than adding commits? I've running into merge conflicts after making an update from my local branch without your changes...

**Comment by abbycross** [comment]
> @abbycross going forward, could you leave comments rather than adding commits? I've running into merge conflicts after making an update from my local branch without your changes...

Yes of course -- apologies, it sounded from your comment yesterday like you were done :)

**Comment by psschwei** [comment]
no worries, always an adventure with github :smile:

**Comment by psschwei** [comment]
> todos:
> 
>     * [ ]  remove `QuantumServerless` and replace by the provider
> 
>     * [ ]  add upload step to the example
> 
>     * [ ]  update `Program` with final name (pending)
> 
>     * [ ]  update `program.py` to reflect final name (pending)
> 
> 

@jenglick I believe these have all been addressed now

File: docs/run/quantum-serverless.mdx
Comment (jyu00) [inline docs/run/quantum-serverless.mdx:8]: Since we are not bringing in Quantum Serverless docs (yet), we should link it and tell people to go there for more info.

File: docs/run/quantum-serverless.mdx
Comment (jenglick) [reply docs/run/quantum-serverless.mdx:12]: Good catch :) @psschwei let's use Sampler

Comment (jyu00) [reply docs/run/quantum-serverless.mdx:12]: Actually can we change this all to use Estimator? Because 1) Estimator is far more popular and 2) Jay is thinking about killing Sampler lol

Comment (jenglick) [reply docs/run/quantum-serverless.mdx:12]: Certainly :) 

We could use this example from the docs: https://learning.quantum-computing.ibm.com/tutorial/working-with-the-qiskit-runtime-estimator-primitive#3-initialize-the-qiskit-runtime-estimator

File: docs/run/quantum-serverless.mdx
Comment (jenglick) [inline docs/run/quantum-serverless.mdx:18]: should be "computing the expectation value using the Qiskit Runtime Estimator primitive" since we've switched away from Sampler.

File: docs/run/quantum-serverless.mdx
Comment (jenglick) [inline docs/run/quantum-serverless.mdx:61]: just checking if this is the right path for this example

**Review by jenglick — COMMENTED** [review]
@psschwei

File: docs/run/quantum-serverless.mdx
Comment (psschwei) [reply docs/run/quantum-serverless.mdx:61]: yes, should be... we're assuming the script is what's noted in the comment in `source_files/my_qiskit_pattern.py` in https://github.com/Qiskit/documentation/blob/0131c29dbffe4d6a662104ba93f8a13ea1df1390/docs/run/quantum-serverless.mdx#L21

**Comment by abbycross** [comment]
Small thing @jenglick @psschwei: is Quantum Serverless supposed to be capitalized, or lowercase? In other words -- is it "branded" and therefore should be capitalized? I see it both ways in this doc.

**Comment by jenglick** [comment]
> Small thing @jenglick @psschwei: is Quantum Serverless supposed to be capitalized, or lowercase? In other words -- is it "branded" and therefore should be capitalized? I see it both ways in this doc.

Let's capitalize to be consistent with, e.g., this: https://www.ibm.com/quantum/summit

**Comment by psschwei** [comment]
I'm going to move this out of draft status now, but it probably shouldn't merge until @jenglick gives the ok

**Comment by abbycross** [comment]
> I'm going to move this out of draft status now, but it probably shouldn't merge until @jenglick gives the ok

Before merging we need a final call on whether to display this page as a child of a section in the table of contents, or whether to make it a standalone page. If we keep it as it is now, we will have to rename either the page or the headings in the toc, since "About quantum serverless" doesn't match the page title "Run workloads with Quantum Serverless". 

I propose making it a standalone page called "Run workloads with Quantum Serverless" with no associated parent section. If you agree I'm happy to fuss with the json file to fix it accordingly

**Comment by jenglick** [comment]
Yes I think it should be a standalone page on the "run" section of the docs called "Run with Quantum Serverless" to be consistent with "Run with primitives" page there. Maybe after "Manage jobs" and before "Hardware" sections?

**Comment by jenglick** [comment]
> Yes I think it should be a standalone page on the "run" section of the docs called "Run with Quantum Serverless" to be consistent with "Run with primitives" page there. Maybe after "Manage jobs" and before "Hardware" sections?

@javabster

**Comment by abbycross** [comment]
@javabster is out sick so I'll make the proposed change, and we can change it later if need be!

**Comment by abbycross** [comment]
Once this is merged to the main branch, someone from the docs team needs to work some magic with syncing the open source in order to publish it. Let me know when the merging is complete and either @beckykd or I can try to publish by tonight!

**Review by abbycross — APPROVED** [review]
Looks good to me :)

File: docs/run/_toc.json
Comment (psschwei) [reply docs/run/_toc.json:59]: done in https://github.com/Qiskit/documentation/pull/209/commits/1c4c91702e0880b576387e9fc50375446642c764

File: docs/run/quantum-serverless.mdx
Comment (psschwei) [reply docs/run/quantum-serverless.mdx]: maybe we split the difference and say something like "authenticate... with your IBM Quantum token, which can be obtained from your IBM Quantum account" (with token linking to the docs and account linking to the personal account page? or is that overcomplicating things?

Comment (abbycross) [reply docs/run/quantum-serverless.mdx]: I'm good with that!

**Comment by psschwei** [comment]
@abbycross could you take one final look to make sure everything looks ok ? (had a bit of funniness with some of the updates, should be all resolved, but better to double check...)

**Comment by jenglick** [comment]
@abbycross can you confirm this is still going to be the URL if we make this a standalone page?

https://docs.quantum-computing.ibm.com/run/quantum-serverless

**Comment by abbycross** [comment]
> @abbycross can you confirm this is still going to be the URL if we make this a standalone page?
> 
> https://docs.quantum-computing.ibm.com/run/quantum-serverless

Confirmed — the url won’t change. This is still the correct one.

**Comment by abbycross** [comment]
> @abbycross could you take one final look to make sure everything looks ok ? (had a bit of funniness with some of the updates, should be all resolved, but better to double check...)

I will take a look in a few moments—I’d like to view the local preview on my laptop but am away from it at the moment. Will let you know how it looks!

**Comment by abbycross** [comment]
Ok -- looks good on my end!! @psschwei

**Comment by psschwei** [comment]
Awesome! :partying_face: 

per https://github.com/Qiskit/documentation/pull/209#issuecomment-1789633345, I think we're ready for merge whenever

**Comment by jenglick** [comment]
@abbycross are you the one who needs to merge this?

**Comment by abbycross** [comment]
@psschwei go ahead and merge — no need for me to do it :) I’ll start the publishing process once that’s done

**Comment by psschwei** [comment]
Think it has to be you @abbycross ... I don't have permissions on this repo

image

**Comment by abbycross** [comment]
Strange, it should allow you since you opened the PR and I've approved -- but no worries, happy to do it!

**Pull request merged**
